### PR TITLE
Add E2E tests for merchant multi-currency setup

### LINKED ISSUE #7382: Merchant facing: Multi-currency setup
As described and assigned in paJDYF-7q7-p2, Sigma needs to improve the E2E test coverage of the multi currency feature. 

The fix for this issue will cover the tests for admin screens that are used to set up multi-currency.

**Possible scenarios:**
- Merchant can enable/disable the multi currency feature.
- Merchant can enable/disable currencies.
- Merchant can enter single currency settings, and:
    - Can change the currency rate manually.
    - Can change the charm price manually.
    - Can change the rounding precision manually.
    - Sees the correct decimal points for these currencies:
       - JPY: 0
       - JOD: 3
       - EUR: 2
- Merchant can see those modifications are applied on the product page.

The dev can decide which ones are good enough to be tested, and dismiss others, and they can add their own tests which may be missing in this list.

### PR DESCRIPTION
**Pull request by rafaelzaleski**
Closes #7382 

#### Changes proposed in this Pull Request

Implement E2E tests for multi-currency setup functionality, including:
- Activate multi-currency support.
- Deactivate multi-currency support.
- Change single currency rate, price rounding, and charm pricing.
- Currency shows the correct number of decimal points.

#### Testing instructions

1. Set up a clean E2E local environment.
2. Execute `npm run test:e2e merchant-multi-currency-setup.spec.js`.
3. Confirm that the tests pass in your local environment.
4. Verify that all E2E tests pass in the GitHub (GH) checks.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7918` or branch name `e2e/7382-spec-merchant-multi-currency-setup` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=e2e/7382-spec-merchant-multi-currency-setup) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5ea27cfa8d6ee8bb7c1c49191f9a9e0f8b80b893
- Build time: 2023-12-18 19:58:48 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/e2e/specs/wcpay/merchant/merchant-admin-multi-currency-setup.spec.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+			expect(
+				parseFloat( beaniePriceOnCurrency )
+					.toFixed( testData.currencyPrecision )
+					.toString()
+			).toBe(
+				( parseFloat( beanieRegularPrice ) * testData.rate )
+					.toFixed( testData.currencyPrecision )
+					.toString()
+			);
```
Comment (cesarcosta99) [inline tests/e2e/specs/wcpay/merchant/merchant-admin-multi-currency-setup.spec.js]: Nit: `toFixed()` [returns](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed#return_value) a string, so a `toString()` call after doesn't change anything, and we can remove it. There are other occurrences below as well.

File: tests/e2e/utils/helpers.js [hunk 2]
```diff
@@ -32,3 +32,51 @@ export const checkPageExists = async ( slug ) => {
...
+			priceElement = document.querySelector(
+				'.woocommerce-Price-amount.amount'
+			);
+		}
+
+		if ( priceElement ) {
+			let priceText = priceElement.textContent || '';
+
+			// Remove non-numeric characters (excluding the decimal point)
+			priceText = priceText.replace( /[^0-9.]/g, '' );
```
Comment (cesarcosta99) [inline tests/e2e/utils/helpers.js:68]: Do you think worths adding the thousand separator (`,`) to this regex?

**Review by cesarcosta99 — APPROVED** [review]
Code looks good and tests well. I left a couple of comments but approving since they're really minor.

- e2e tests pass locally ✅ 
- e2e tests pass in the CI ✅

**Review by cesarcosta99 — APPROVED** [review]
Code looks good and tests well. I left a couple of comments but approving since they're really minor.

- e2e tests pass locally ✅ 
- e2e tests pass in the CI ✅

File: tests/e2e/specs/wcpay/merchant/merchant-admin-multi-currency-setup.spec.js (hunk 1 above)
Comment (rafaelzaleski) [reply tests/e2e/specs/wcpay/merchant/merchant-admin-multi-currency-setup.spec.js]: Good catch. Fixed on https://github.com/Automattic/woocommerce-payments/pull/7918/commits/b257f8b225fcedbbd66eb4e0f76542de4bf65008

File: tests/e2e/utils/helpers.js (hunk 2 above)
Comment (rafaelzaleski) [reply tests/e2e/utils/helpers.js:68]: Not required. The only concern would be if the decimal separator is a comma, but for our current test cases, this approach should suffice.

**Comment by rafaelzaleski** [comment]
I'll proceed with merging this PR despite the errors in the JS tests check, as they seem to be [linked to a different PR](https://github.com/Automattic/woocommerce-payments/pull/7851#issuecomment-1861694785).

**Pull request merged**